### ISSUE TITLE
Annotations tab fails to update when errors are thrown due to over-eager attempts to translate URLs

### ISSUE THREAD

**Issue by cmungall** [protegeproject/protege#444]
I am using the latest Protege ('About' reports 5.0.0, Build beta-24 which is confusing)

The following two lines have a literal value whose value is a badly formatted URL:

https://github.com/AgriculturalSemantics/agro/blob/0af9b5fe6db92e8357388e1d9b7d20ced2ca4c56/src/ontology/agro-edit.owl#L36-37

Note that the value is not typed to xsd:anyURI, so technically this is valid OWL, even if the insertion of a newline is unintentional.

It appears that Protege attempts to treat this as a URL (potentially helpful, as there may be many 'cryptic' URLs out there as literals/strings), presumably by sniffing the string. However, after sniffing it fails to convert to a URL, due to the newline, and throws the error below.

This appears to have the knock on effect of the Annotations tab not being updated, leading to very confusing behavior when trying to look at the class in question.

The error can be reproduced using the agro-edit file from commit 0af9b5fe6db92e8357388e1d9b7d20ced2ca4c56

```
Uncaught Exception in thread 'AWT-EventQueue-0'
java.lang.IllegalArgumentException: Illegal character in query at index 49: https://stats.oecd.org/glossary/detail.asp?ID=447

        at java.net.URI.create(URI.java:852) ~[na:1.8.0_40]
        at org.semanticweb.owlapi.model.IRI.toURI(IRI.java:56) ~[na:na]
        at org.protege.editor.owl.ui.renderer.OWLAnnotationCellRenderer2.renderExternalIRI(OWLAnnotationCellRenderer2.java:384) ~[na:na]
        at org.protege.editor.owl.ui.renderer.OWLAnnotationCellRenderer2.renderIRI(OWLAnnotationCellRenderer2.java:344) ~[na:na]
        at org.protege.editor.owl.ui.renderer.OWLAnnotationCellRenderer2.renderLiteral(OWLAnnotationCellRenderer2.java:464) ~[na:na]
        at org.protege.editor.owl.ui.renderer.OWLAnnotationCellRenderer2.access$200(OWLAnnotationCellRenderer2.java:35) ~[na:na]
        at org.protege.editor.owl.ui.renderer.OWLAnnotationCellRenderer2$1.visit(OWLAnnotationCellRenderer2.java:322) ~[na:na]
        at org.protege.editor.owl.ui.renderer.OWLAnnotationCellRenderer2$1.visit(OWLAnnotationCellRenderer2.java:312) ~[na:na]
        at uk.ac.manchester.cs.owl.owlapi.OWLLiteralImplPlain.accept(OWLLiteralImplPlain.java:213) ~[na:na]
        at org.protege.editor.owl.ui.renderer.OWLAnnotationCellRenderer2.renderAnnotationValue(OWLAnnotationCellRenderer2.java:312) ~[na:na]
        at org.protege.editor.owl.ui.renderer.OWLAnnotationCellRenderer2.renderCellValue(OWLAnnotationCellRenderer2.java:196) ~[na:na]
        at org.protege.editor.owl.ui.renderer.OWLAnnotationCellRenderer2.renderCellValue(OWLAnnotationCellRenderer2.java:200) ~[na:na]
        at org.protege.editor.owl.ui.renderer.OWLAnnotationCellRenderer2.fillPage(OWLAnnotationCellRenderer2.java:160) ~[na:na]
        at org.protege.editor.owl.ui.renderer.layout.PageCellRenderer.getListCellRendererComponent(PageCellRenderer.java:93) ~[na:na]
        at org.protege.editor.owl.ui.framelist.OWLFrameListRenderer.getListCellRendererComponent(OWLFrameListRenderer.java:150) ~[na:na]
        at org.protege.editor.core.ui.list.MList$MListCellRenderer.getListCellRendererComponent(MList.java:260) ~[na:na]
        at org.protege.editor.owl.ui.framelist.OWLFrameList$OWLFrameListUI.updateLayoutState(OWLFrameList.java:837) ~[na:na]
        at javax.swing.plaf.basic.BasicListUI.maybeUpdateLayoutState(BasicListUI.java:1311) ~[na:1.8.0_40]
        at javax.swing.plaf.basic.BasicListUI.getPreferredSize(BasicListUI.java:578) ~[na:1.8.0_40]
        at javax.swing.JComponent.getPreferredSize(JComponent.java:1662) ~[na:1.8.0_40]
        at javax.swing.ScrollPaneLayout.layoutContainer(ScrollPaneLayout.java:791) ~[na:1.8.0_40]
        at java.awt.Container.layout(Container.java:1508) ~[na:1.8.0_40]
        at java.awt.Container.doLayout(Container.java:1497) ~[na:1.8.0_40]
        at java.awt.Container.validateTree(Container.java:1693) ~[na:1.8.0_40]
        at java.awt.Container.validateTree(Container.java:1702) ~[na:1.8.0_40]
        at java.awt.Container.validateTree(Container.java:1702) ~[na:1.8.0_40]
        at java.awt.Container.validateTree(Container.java:1702) ~[na:1.8.0_40]
        at java.awt.Container.validateTree(Container.java:1702) ~[na:1.8.0_40]
        at java.awt.Container.validateTree(Container.java:1702) ~[na:1.8.0_40]
        at java.awt.Container.validateTree(Container.java:1702) ~[na:1.8.0_40]
        at java.awt.Container.validateTree(Container.java:1702) ~[na:1.8.0_40]
        at java.awt.Container.validateTree(Container.java:1702) ~[na:1.8.0_40]
        at java.awt.Container.validateTree(Container.java:1702) ~[na:1.8.0_40]
        at java.awt.Container.validateTree(Container.java:1702) ~[na:1.8.0_40]
        at java.awt.Container.validateTree(Container.java:1702) ~[na:1.8.0_40]
        at java.awt.Container.validate(Container.java:1628) ~[na:1.8.0_40]
        at javax.swing.RepaintManager$3.run(RepaintManager.java:704) ~[na:1.8.0_40]
        at javax.swing.RepaintManager$3.run(RepaintManager.java:702) ~[na:1.8.0_40]
```

**Comment by matthewhorridge** [comment]
Hi Chris,

This works for me with the 5.0.0 release.  The malformed URL isn't actually parsed and displayed as a URL:

<img width="676" alt="baduri" src="https://cloud.githubusercontent.com/assets/488607/15801405/50d06c36-2a48-11e6-9525-bfe64c7c7328.png">

The issue you report sounds like a variant of #427 which was fixed for the 5.0.0 release.  I noted that you said 'About' reports you're using beta-24.  When I look at 'About' for the 5.0.0 release I don't see this.  I'm just wondering if you're unexpectedly using a copy of beta-24.  Please could you check this?

**Comment by cmungall** [comment]
You're right, I was totally sure I was running the latest, but user error due to my contorted command line setup.... as you were!
